### ISSUE TITLE
With version 12.2, the dependencies no longer install behind corporate registry

### ISSUE THREAD

**Issue by woppa684** [angular/angular-cli#21529]
# 🐞 Bug report

### Command (mark with an `x`)

- [x] new
- [x] build
- [x] serve
- [x] test
- [x] e2e
- [x] generate
- [x] add
- [x] update
- [x] lint
- [x] extract-i18n
- [x] run
- [x] config
- [x] help
- [x] version
- [x] doc

### Is this a regression?

Yes, it works in the latest 12.1

### Description

The `esbuilder` dependency tries to install its own dependencies in a post-install script and fails behind a corporate proxy / registry. See these (maybe related) issues: https://github.com/evanw/esbuild/issues/1382 https://github.com/evanw/esbuild/issues/1485, apparently this is correct behavior, but now it affects angular CLI as well.

## 🔬 Minimal Reproduction
I cannot create an online repoduction of this.

## 🔥 Exception or Error

<pre><code>
> esbuild@0.12.17 postinstall C:\Repos\myrepo\node_modules\esbuild
> node install.js

Trying to install "esbuild-windows-64" using npm
Failed to install "esbuild-windows-64" using npm: Command failed: npm install --loglevel=error --prefer-offline --no-audit --progress=false esbuild-windows-64@0.12.17
npm ERR! code E401
npm ERR! Unable to authenticate, need: Basic realm="http://ourcorporateregistry", Negotiate, NTLM

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\myuser\AppData\Roaming\npm-cache\_logs\2021-08-09T05_52_58_429Z-debug.log

Trying to download "https://registry.npmjs.org/esbuild-windows-64/-/esbuild-windows-64-0.12.17.tgz"
Failed to download "https://registry.npmjs.org/esbuild-windows-64/-/esbuild-windows-64-0.12.17.tgz": connect ECONNREFUSED 104.16.22.35:443
</code></pre>

## 🌍 Your Environment

<pre><code>
Angular CLI: 12.2.0
Node: 12.21.0
Package Manager: npm 6.14.11
OS: win32 x64

Angular: 12.2.0
... cli, compiler, compiler-cli, core, language-service

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.0
@angular-devkit/build-angular   12.2.0
@angular-devkit/core            12.2.0
@angular-devkit/schematics      12.1.3
@schematics/angular             12.1.3
ng-packagr                      12.2.0
rxjs                            6.6.7
typescript                      4.3.5
</code></pre>

**Anything else relevant?**
64-bit Windows 10 behind a corporate (Azure Artifacts) registry

**Comment by woppa684** [comment]
It might be worth mentioning that we use project-level .npmrc files and that the postinstall script of esbuild tries to use the global .npmrc file (which might be the intended usage?). Anyway, current workaround is to manually copy my project-specific .npmrc file to the global location before running npm install.

**Comment by alan-agius4** [comment]
I believe adding `always-auth = true` to your npmrc file should fix this.

Can you please try?

**Comment by woppa684** [comment]
Yes we already have this, it really is about the fact that we use project-specific .npmrc files and that a postinstall script that just calls npm will use the global .npmrc (as described above). This is probably not something that can be fixed at Angular CLI level so I think I can better close this? It was just a very unpleasant surprise when upgrading to the new CLI. We have had similar things with other libraries trying to download packages by themselves (e.g. node-sass) but there were other workarounds for that.

**Comment by alan-agius4** [comment]
Indeed, I don't see this as being actionable from our end. That said, the fact that it doesn't use the local NPMRC files seem like a different ESbuild bug. 

This is caused because the `CWD` is changed from the project root to a temp directory, which caused the local NPMRC files not to be read by NPM/Yarn.

See: https://github.com/evanw/esbuild/blob/eb2872e1ab4ae09bef6fae71db6eb2381ede131b/lib/npm/install.ts#L178

I also suggest to subscribe to https://github.com/evanw/esbuild/issues/789

**Comment by bgotink** [comment]
For what it's worth, we've "solved" this issue by adding a resolution replacing `esbuild` with the slower `esbuild-wasm`. I don't think npm supports this, but we use yarn which does have this capability ([docs](https://yarnpkg.com/configuration/manifest/#resolutions)).

```json5
"resolutions": {
  // replace 0.12.17 with the version of esbuild your angular version is using
  "esbuild@0.12.17": "npm:esbuild-wasm@0.12.17"
}
```

**Comment by woppa684** [comment]
@alan-agius4 I see that the esbuild bug you mentioned should be solved. Does that also mean that it is also solved in Angular CLI? And it is only solved in v13 then, right? I see 12.2.7 still uses a 0.12.x version of esbuild. If this is the case, would it be possible to put the fixed version in 12.2.8 (if there is still a version coming up)? Upgrading to Angular 13 is not planned soon for our company.
